engine/include/engine/engine.h

```diff
@@ -8,12 +8,34 @@
 struct ResultSet {
 	std::vector<gdf_column *> columns;
 	std::vector<std::string> names;
-	ral::frame::BlazingTableView *blazingTableView;
+	std::unique_ptr<ral::frame::BlazingTable> blazingTable;
+
+	ResultSet() {
+		columns = {};
+		names = {};
+		blazingTable = nullptr;
+	}
+
+	ResultSet(std::unique_ptr<ral::frame::BlazingTable> blazingTableValue)
+		: blazingTable{std::forward<std::unique_ptr<ral::frame::BlazingTable>>(blazingTableValue)} {}
+
+	ResultSet(const ResultSet & other) {
+		columns = std::move(other.columns);
+		names = std::move(other.names);
+		blazingTable.swap(const_cast<ResultSet &>(other).blazingTable);
+	}
+
+	ResultSet & operator=(const ResultSet & other) {
```

Comment:
this is copy assignment operator not a move operator.  use ```operator = (const ResultSet && other)``` if you are tryign to define a move operator

Reply:
I'm adding the assign operator beacuse it's a restriction of cython. Internally, cython generates cpp code that uses copy constructor and assign operator.